### ISSUE TITLE
Add /simplehash-api #2

### ISSUE THREAD

**Issue by westonnelson** [NFTEarth/nftearth-exchangev1#3]
# Description
We are working on integrating the Simplehash API on Optimism, and then Arbitrum. We have had issues receives offers/listings so far with the data working as would like. We need listings by NFT and additionally bids and offers by Collection and NFT

# Existing
We had tried Simplehash before, but ran into issues getting the API to get all the data required for a high-performance marketplace.

# References
https://docs.simplehash.com/reference/collections-by-marketplace

# Security
Discuss internally best practices here.

# Tasks
 -Research Simplehash documentation completely
 -Build demos using webhook API
 -Write tests
 -If able to confirm all works as providers say it should, deploy

**Comment by westonnelson** [comment]
Using this specific CURL request below, we've been able to confirm a response is being returned with accurate real-time data for an Optimism NFT Collection, Bored Town. However, it is continuing to fail using the API key for the DAO, in addition to missing a couple of data points we require. See the request and response below:


CURL:

curl --request GET \
     --url https://api.simplehash.com/api/v0/nfts/optimism/0x8E56343adAFA62DaC9C9A8ac8c742851B0fb8b03/2620 \
     --header 'X-API-KEY: sh_sk1_Z4jhWXXBE09em' \
     --header 'accept: application/json'

RESPONSE:

{
  "nft_id": "optimism.0x8e56343adafa62dac9c9a8ac8c742851b0fb8b03.2620",
  "chain": "optimism",
  "contract_address": "0x8E56343adAFA62DaC9C9A8ac8c742851B0fb8b03",
  "token_id": "2620",
  "name": "Bored Town #2620",
  "description": "Bored Town is a collection of 5555 Bored Town Monsters living on the Optimism blockchain. As an open-source brand (aka CC0), Bored Town holders have the chance to create whatever they put their mind to, both for personal and commercial purposes.",
  "image_url": "https://cdn.simplehash.com/assets/33b850de2f56be16ebe349f1a620eb2781ed2fcadf3261b9d78bd29483f16bc8.png",
  "video_url": null,
  "audio_url": null,
  "model_url": null,
  "previews": {
    "image_small_url": "https://lh3.googleusercontent.com/21Wsd8VbvfN9G8RevLcTGMUxG0QgqbR_uVkAitYKc1IHJlReh72Shpwrns5O9PeZF6SH-2GyPqp45LAxLd6FoZPOF3Q4jAAMig3S=s250",
    "image_medium_url": "https://lh3.googleusercontent.com/21Wsd8VbvfN9G8RevLcTGMUxG0QgqbR_uVkAitYKc1IHJlReh72Shpwrns5O9PeZF6SH-2GyPqp45LAxLd6FoZPOF3Q4jAAMig3S",
    "image_large_url": "https://lh3.googleusercontent.com/21Wsd8VbvfN9G8RevLcTGMUxG0QgqbR_uVkAitYKc1IHJlReh72Shpwrns5O9PeZF6SH-2GyPqp45LAxLd6FoZPOF3Q4jAAMig3S=s1000",
    "image_opengraph_url": "https://lh3.googleusercontent.com/21Wsd8VbvfN9G8RevLcTGMUxG0QgqbR_uVkAitYKc1IHJlReh72Shpwrns5O9PeZF6SH-2GyPqp45LAxLd6FoZPOF3Q4jAAMig3S=k-w1200-s2400-rj",
    "blurhash": "UIOzGTt7%$V@?^V@H?og8_of%#WBXmayr=kC"
  },
  "background_color": null,
  "external_url": "https://quixotic.io/collection/boredtown",
  "created_date": "2022-06-11T15:11:27",
  "status": "minted",
  "token_count": 1,
  "owner_count": 1,
  "owners": [
    {
      "owner_address": "0xD131F1BcDd547e067Af447dD3C36C99d6be9FdEB",
      "quantity": 1,
      "first_acquired_date": "2023-01-19T02:41:37",
      "last_acquired_date": "2023-01-19T02:41:37"
    }
  ],
  "last_sale": {
    "from_address": "0x6C991016b4588B3c73A984b87275f8FF8Ea72967",
    "to_address": "0xD131F1BcDd547e067Af447dD3C36C99d6be9FdEB",
    "quantity": 1,
    "timestamp": "2023-01-19T02:41:37",
    "transaction": "0xbcd1d76af59e4a1b559d515d725ddbdf027633fdd854b79beeec4ef825705694",
    "marketplace_id": "opensea",
    "marketplace_name": "OpenSea",
    "is_bundle_sale": false,
    "payment_token": {
      "payment_token_id": "optimism.native",
      "name": "Ether",
      "symbol": "ETH",
      "address": null,
      "decimals": 18
    },
    "unit_price": 17000000000000000,
    "total_price": 17000000000000000
  },
  "contract": {
    "type": "ERC721",
    "name": "Bored Town",
    "symbol": null
  },
  "collection": {
    "collection_id": "28ad42cc5589cf4363a6e6dc3bc3b7ba",
    "name": "Bored Town",
    "description": "\"Bored Town is a collection of 5555 Bored Town Monsters living on the Optimism blockchain. Created by sorryvrerror a Thai NFT artist. Bored Town holders have the chance to create derivative work whatever they put their mind to both for personal and commercial purposes under CC BY-SA terms. Holders please join https://guild.xyz/boredtown to get your role on Bored Town Discord and be eligible for giveaway raffles.\"",
    "image_url": "https://lh3.googleusercontent.com/_RISPyvwULVy5haZ23WIj1rk4A9FKxCnZJ0qbbGRvI-O23EA1iTgepcCxyiaQOE0AqB3qiNvo5oz_7xoqB2npQlfMtJedVElXvQ",
    "banner_image_url": "https://lh3.googleusercontent.com/S4l29s3AVwEJcTBg5sxCndvDRrvww5qbfqlDFG3BSQanFs-YZ1Ue7tLi9yynaPWKBdbUnl4ztPvCD1EpbDNEkx23M4XPkXsv_UU=w2500",
    "external_url": "https://boredtown.app/nfts",
    "twitter_username": "BoredTownNFT",
    "discord_url": "https://discord.gg/boredtown",
    "marketplace_pages": [
      {
        "marketplace_id": "opensea",
        "marketplace_name": "OpenSea",
        "marketplace_collection_id": "boredtown",
        "nft_url": "https://opensea.io/assets/optimism/0x8e56343adafa62dac9c9a8ac8c742851b0fb8b03/2620",
        "collection_url": "https://opensea.io/collection/boredtown",
        "verified": false
      }
    ],
    "metaplex_mint": null,
    "metaplex_first_verified_creator": null,
    "floor_prices": [
      {
        "marketplace_id": "opensea",
        "marketplace_name": "OpenSea",
        "value": 16500000000000000,
        "payment_token": {
          "payment_token_id": "optimism.native",
          "name": "Ether",
          "symbol": "ETH",
          "address": null,
          "decimals": 18
        }
      }
    ],
    "distinct_owner_count": 3784,
    "distinct_nft_count": 5554,
    "total_quantity": 5554,
    "top_contracts": [
      "optimism.0x8e56343adafa62dac9c9a8ac8c742851b0fb8b03"
    ]
  },
  "extra_metadata": {
    "attributes": [
      {
        "trait_type": "Background",
        "value": "Light Gray"
      },
      {
        "trait_type": "Accessory",
        "value": "Three Eyes"
      },
      {
        "trait_type": "Body",
        "value": "Cow Cat With Tuxido And Wings"
      },
      {
        "trait_type": "Head",
        "value": "Cabbit"
      }
    ],
    "compiler": "nft-inator.com",
    "image_original_url": "ipfs://QmeqU8hKaSpJ8dzNbfxD2tnZ5CDUb83ByAPFbcp9J71HQK",
    "animation_original_url": null,
    "metadata_original_url": "ipfs://QmZ2eKEiuV1UKxgsLrPUK9JXhvLhtbetxUwZEHRiqUaHq4/2620.json"
  }
}

**Comment by westonnelson** [comment]
Update: Met with the SimpleHash team; have obtained an Enterprise SLA for use with the API Key through SimpleHash. Looking quite promising. 

Need to ensure that the data being returned is thorough enough to use now in the marketplace to create a Quix-like UI though, while the team at SimpleHash is in the process of finishing full support for both Layer2 networks we aim to deploy on.

**Comment by westonnelson** [comment]
SimpleHash integration is underway. We are able use the Transfers endpoint to retrieve Sales History and we've successfully been able to add tokens owned by address, this is currently already being reflected in the UI. In addition, SimpleHash has confirmed they will be adding support for the missing endpoint we need in Q1 2023 on both Optimism and Arbitrum. Closing this issue.

Screencap attached of an example successful response of Optimism transfers.
![TransfersOptimism](https://user-images.githubusercontent.com/29180454/213762460-4d06134b-dd3b-44cf-b43f-4f0b56f7ae8a.png)
